File: android/src/main/kotlin/uz/shs/video_player/models/Movie.kt
package uz.shs.video_player.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Movie(
    @SerializedName("id")
    val id: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("description")
    val description: String,
    @SerializedName("image")
    val image: String,
    @SerializedName("duration")
    val duration: Long,
    @SerializedName("resolutions")
    var resolutions: HashMap<String, String>
) : Serializable

File: lib/models/movie.dart
class Movie {
  Movie({
    required this.id,
    required this.title,
    required this.description,
    required this.image,
    required this.duration,
    required this.resolutions,
  });

  final String id;
  final String title;
  final String description;
  final String image;
  final int duration;
  final Map<String, String> resolutions;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['id'] = id;
    map['title'] = title;
    map['description'] = description;
    map['image'] = image;
    map['duration'] = duration;
    map['resolutions'] = resolutions;
    return map;
  }

  @override
  String toString() =>
      'Movie{id: $id, title: $title, description: $description, image: $image, duration: $duration, resolutions: $resolutions}';
}
